packages/core/src/parseAmplifyGen2.ts

```diff
@@ -0,0 +1,267 @@
+		}
+
+		if (!config.Auth.Cognito.loginWith) {
+			delete config.Auth.Cognito.loginWith;
+		}
+	}
+
+	if (gen2Config.analytics) {
+		const { amazon_pinpoint } = gen2Config.analytics;
+
+		if (amazon_pinpoint) {
+			const Pinpoint = {
+				appId: amazon_pinpoint.app_id,
+				region: amazon_pinpoint.aws_region,
+			};
+			if (!config.Analytics) {
+				config.Analytics = {
+					Pinpoint,
+				};
+			} else {
+				config.Analytics!.Pinpoint = Pinpoint;
+			}
+		}
+	}
+
+	if (gen2Config.geo) {
+		const { aws_region, geofence_collections, maps, search_indices } =
+			gen2Config.geo;
+		config.Geo = {
+			LocationService: {
+				region: aws_region,
+				searchIndices: search_indices,
+				geofenceCollections: geofence_collections,
+				maps,
+			},
+		};
+	}
+
+	if (gen2Config.data) {
+		const {
+			aws_region,
+			default_authorization_type,
+			url,
+			api_key,
+			model_introspection,
+		} = gen2Config.data;
+
+		const GraphQL: APIGraphQLConfig = {
+			endpoint: url,
+			defaultAuthMode: getGraphQLAuthMode(default_authorization_type),
+			region: aws_region,
+			apiKey: api_key,
+			modelIntrospection: model_introspection,
+		};
+
+		if (!config.API) {
+			config.API = {
+				GraphQL,
+			};
+		} else {
+			config.API.GraphQL = GraphQL;
+		}
+	}
+
+	if (gen2Config.notifications) {
+		const { aws_region, channels, pinpoint_app_id } = gen2Config.notifications;
+
+		if (channels.in_app_messaging) {
+			const InAppMessaging = {
+				Pinpoint: {
+					appId: pinpoint_app_id,
+					region: aws_region,
+				},
+			};
+			if (!config.Notifications) {
+				config.Notifications = {
+					InAppMessaging,
+				};
+			} else {
+				config.Notifications.InAppMessaging = InAppMessaging;
+			}
+		}
+		if (channels.apns || channels.fcm) {
+			const PushNotification = {
+				Pinpoint: {
+					appId: pinpoint_app_id,
+					region: aws_region,
+				},
+			};
+
+			if (!config.Notifications) {
+				config.Notifications = {
+					PushNotification,
+				};
+			} else {
+				config.Notifications.PushNotification = PushNotification;
+			}
+		}
+	}
+
+	return config;
+}
+
+function getGraphQLAuthMode(authType: AuthType): GraphQLAuthMode {
+	switch (authType) {
+		case 'AMAZON_COGNITO_USER_POOLS':
+			return 'userPool';
+		case 'API_KEY':
+			return 'apiKey';
+		case 'AWS_IAM':
+			return 'iam';
+		case 'AWS_LAMBDA':
+			return 'lambda';
+		case 'OPENID_CONNECT':
+			return 'oidc';
+		default:
+			throw new Error('Invalid AuthMode configured');
+	}
+}
+
+function getOAuthProviders(
+	providers?: Gen2OAuthIdentityProviders[],
+): OAuthProvider[] {
+	const oauthProviders: OAuthProvider[] = [];
+	providers?.forEach(provider => {
+		if (provider === 'GOOGLE') {
+			oauthProviders.push('Google');
+		}
+		if (provider === 'LOGIN_WITH_AMAZON') {
+			oauthProviders.push('Amazon');
+		}
+		if (provider === 'FACEBOOK') {
+			oauthProviders.push('Facebook');
+		}
+		if (provider === 'SIGN_IN_WITH_APPLE') {
+			oauthProviders.push('Apple');
+		}
+	});
+
+	return oauthProviders;
+}
```

Comment:
Why not use `Array.map` and simplify the lookup logic by assign an object with the IdentityProvider values mapped to their ResourcesConfig name?
```suggestion
const providerNames: Record<Gen2OAuthIdentityProviders, OAuthProvider> = {
	GOOGLE: 'Google',
	LOGIN_WITH_AMAZON: 'Amazon',
	FACEBOOK: 'Facebook',
	SIGN_IN_WITH_APPLE: 'Apple',
};
function getOAuthProviders(
	providers?: Gen2OAuthIdentityProviders[],
): OAuthProvider[] {
	return (providers ?? []).map(provider => providerNames[provider]);
}
```

Reply:
Nice suggestion, I would do this!